tests/C-tests/element_multiply_matrix_typed.c

```diff
@@ -0,0 +1,226 @@
+#include "bml.h"
+#include "../macros.h"
+#include "../typed.h"
+#include "ellblock/bml_allocate_ellblock.h"
+#include "bml_utilities.h"
+
+#include <complex.h>
+#include <math.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#ifdef _OPENMP
+#include<omp.h>
+#else
+#include<time.h>
+#endif
+
+static void TYPED_FUNC(
+    ref_element_multiply) (
+    const int N,
+    const REAL_T * A,
+    const REAL_T * B,
+    REAL_T * C,
+    const double alpha,
+    const double beta,
+    const double threshold)
+{
+    for (int i = 0; i < N; i++)
+    {
+        for (int j = 0; j < N; j++)
+        {
+            C[ROWMAJOR(i, j, N, N)] *= beta;
+            C[ROWMAJOR(i, j, N, N)] +=
+                alpha * A[ROWMAJOR(i, j, N, N)] * B[ROWMAJOR(i, j, N, N)];
+        }
+    }
+}
+
+#if defined(SINGLE_REAL) || defined(SINGLE_COMPLEX)
+#define ABS_TOL 2e-6
+#else
+#define ABS_TOL 1e-12
+#endif
+
+static int TYPED_FUNC(
+    compare_matrix) (
+    const int N,
+    const bml_matrix_precision_t matrix_precision,
+    REAL_T * A,
+    REAL_T * B)
+{
+    int max_row = MIN(N, PRINT_THRESHOLD);
+    int max_col = MIN(N, PRINT_THRESHOLD);
+
+    LOG_INFO("Compare matrices\n");
+    for (int i = 0; i < N * N; i++)
+    {
+        if (ABS(A[i] - B[i]) > ABS_TOL)
+        {
+            LOG_INFO("First matrix\n");
+            bml_print_dense_matrix(N, matrix_precision, dense_row_major, A, 0,
+                                   max_row, 0, max_col);
+            LOG_INFO("Second matrix\n");
+            bml_print_dense_matrix(N, matrix_precision, dense_row_major, B, 0,
+                                   max_row, 0, max_col);
+            LOG_INFO("element %d outside %1.2e\n", i, ABS_TOL);
+#if defined(SINGLE_COMPLEX) || defined(DOUBLE_COMPLEX)
+            LOG_INFO("A[%d] = %e+%ei\n", i, creal(A[i]), cimag(A[i]));
+            LOG_INFO("B[%d] = %e+%ei\n", i, creal(B[i]), cimag(B[i]));
+            LOG_INFO("abs(A[%d]-B[%d]) = %e\n", i, i, ABS(A[i] - B[i]));
+#else
+            LOG_INFO("A[%d] = %e\n", i, A[i]);
+            LOG_INFO("B[%d] = %e\n", i, B[i]);
+            LOG_INFO("abs(A[%d]-B[%d]) = %e\n", i, i, ABS(A[i] - B[i]));
+#endif
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void TYPED_FUNC(
+    setup_bsizes2) (
```

Comment:
what is this function for?

Reply:
set block size, copied from setup_bsize in multiply_matrix_type.c.

Reply:
will be removed in rebasing this commit.